js/src/leaflet.js

```diff
@@ -14,49 +16,49 @@ require('leaflet.awesome-markers');
 require('leaflet-search');
 
 // Monkey patch GridLayer for smoother URL updates
-L.patchGridLayer = function(layer) {
-  layer._refreshTileUrl = function(tile, url) {
+L.patchGridLayer = function (layer) {
+  layer._refreshTileUrl = function (tile, url) {
     //use a image in background, so that only replace the actual tile, once image is loaded in cache!
     var img = new Image();
-    img.onload = function() {
-      L.Util.requestAnimFrame(function() {
+    img.onload = function () {
+      L.Util.requestAnimFrame(function () {
         tile.el.src = url;
       });
     }
     img.src = url;
   };
 
-  layer.refresh = function() {
+  layer.refresh = function () {
     //prevent _tileOnLoad/_tileReady re-triggering a opacity animation
     var wasAnimated = this._map._fadeAnimated;
     this._map._fadeAnimated = false;
 
     for (var key in this._tiles) {
-      tile = this._tiles[key];
+      var tile = this._tiles[key];
       if (tile.current && tile.active) {
         var oldsrc = tile.el.src;
         var newsrc = this.getTileUrl(tile.coords);
         if (oldsrc != newsrc) {
           //L.DomEvent.off(tile, 'load', this._tileOnLoad); ... this doesnt work!
-          this._refreshTileUrl(tile,newsrc);
+          this._refreshTileUrl(tile, newsrc);
         }
       }
     }
 
     if (wasAnimated) {
-      setTimeout(function() { map._fadeAnimated = wasAnimated; }, 5000);
+      setTimeout(function () {this._map._fadeAnimated = wasAnimated;}, 5000);
```

Comment:
I don't understand why this code was working before.

But I am also not sure it is still working now. Shouldn't this function be an arrow function?

Reply:
Yeah I don't know why this is inside a timeout....

Reply:
I don't remember the reason for the timeout. I remember this code was a bit weak but that was the only thing we could do.

Let's make it an arrow function anyway so the code still works, otherwise `this` is not defined properly.

Reply:
See https://github.com/jupyter-widgets/ipyleaflet/pull/495 for reference

Reply:
Actually I feel like this timeout thing never worked. Maybe this code is actually never triggered. Oo'

Reply:
I'd say let's just leave it there until there is a PR for cleaning JS specific code.